tests/foreman/api/test_repository.py

```diff
@@ -639,8 +641,45 @@ def test_negative_create_checksum(self):
 
         :CaseImportance: Critical
         """
-        with self.assertRaises(HTTPError):
-            entities.Repository(checksum_type=gen_string('alpha')).create()
+        for downloadpolicy in 'background', 'immediate':
```

Comment:
same here, why do you put different functionality there?

Reply:
@oshtaier earlier the test was creating the repo with default download_policy (on_demand) but not covering the other type of download_policies i.e. background/immediate hence added the same to cover more on it.
and we added the nagative test for on_demand separately to cover on that type of download policy .

Do you think here should be two test one per download policy background/immediate?